crates/dbsp/src/trace/spine_fueled.rs

```diff
@@ -762,18 +767,20 @@ where
         self.dirty
     }
 
-    fn truncate_values_below(&mut self, lower_bound: &Self::Val) {
-        // Set the bound, we postpone the actual GC till when we merge batches
-        // in the trace.
-        self.lower_val_bound = Some(if let Some(bound) = &self.lower_val_bound {
-            max(bound, lower_bound).clone()
-        } else {
-            lower_bound.clone()
-        });
+    fn retain_keys(&mut self, filter: Filter<Self::Key>) {
```

Comment:
I find this method name misleading. It really sets the filter for the keys.

Reply:
This is by analogy with the standard library, which has many `retain` methods for various collection types with similar semantics.

Reply:
but this method doesn't seem to do that!

Reply:
It tells the collection to retain elements that satisfy the predicate, which is exactly what `Vec::retain` does for example.